### ISSUE TITLE
MD034 incorrectly checks URLs inside code blocks

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#230]
## Bug Description
MD034 (no-bare-urls) incorrectly flags URLs that appear inside code blocks (both fenced and indented).

## Impact
- ~2,700 false positives when running on markdownlint test suite
- Affects any documentation with code examples containing URLs

## Example
```markdown
Code block with URL:
```yaml
ruleInformation: 'https://github.com/example/url'  # Should NOT be flagged
```

Indented code block:
    https://example.com  # Should NOT be flagged
```

## Solution
Similar to MD007 fix in #228:
1. Convert MD034 from `Rule` to `AstRule` trait
2. Add code block detection using AST
3. Skip lines within code block ranges

## Test Cases
- [ ] URLs in fenced code blocks should be ignored
- [ ] URLs in indented code blocks should be ignored
- [ ] URLs outside code blocks should still be detected

## Related
- Parent issue: #229
- Similar fix: #228 (MD007)
